pydeseq2/utils.py

```diff
@@ -1608,3 +1608,64 @@ def lowess(
         delta = (1 - delta**2) ** 2
 
     return yest
+
+
+def make_rle_plot(
+    count_matrix: np.array,
+    sample_ids: np.array,
+    normalize: bool = False,
+    save_path: Optional[str] = None,
+    **kwargs,
+) -> None:
+    """
+    Create a ratio of log expression plot using matplotlib.
+
+    Parameters
+    ----------
+    count_matrix : ndarray
+        An mxn matrix of count data, where m is the number of samples (rows),
+        and n is the number of genes (columns).
+
+    sample_ids : ndarray
+        An array of sample identifiers.
+
+    normalize : bool
+        Whether to normalize the count matrix before plotting. (default: ``False``).
+
+    save_path : str or None
+        The path where to save the plot. If left None, the plot won't be saved
+        (default: ``None``).
+
+    **kwargs :
+        Additional keyword arguments passed to matplotlib's boxplot function.
+    """
+    if normalize:
+        print("Plotting normalized RLE plot...")
+        geometric_mean = np.exp(np.mean(np.log(count_matrix + 1), axis=0))
+        size_factors = np.median(count_matrix / geometric_mean, axis=1)
+        count_matrix = count_matrix / size_factors[:, np.newaxis]
```

Comment:
This is recomputing median-of-ratios size factors, right? (up to the +1 in the log)

In that case, it might be better to add a `size_factors` argument to `make_rle_plot`. 

From there, in `dds.plot_rle`, if `normalize=True`, we would first check whether size factors (`self.obsm["size_factors"]`) were already computed. If so, we pass them directly. If not, we call `self.fit_size_factors` first.

Reply:
Hi @BorisMuzellec ,

For some reason when I use the size factors computed in dds.fit_size_factors() I do not get an RLE plot with the sample medians centered around 0.

![image](https://github.com/user-attachments/assets/feee1a9d-eabe-4bf8-ae08-54e2cb6c31ad)

But when I compute the sizefactors internally I do get an RLE plot with the sample medians centered around 0

![image](https://github.com/user-attachments/assets/4bec330e-5bc0-40ee-989b-610fdc9f4c1e)

Do you know what could be causing this?